File: v1-java-vs-ruby_r/Ueberlagert.rb
class Ueberlagert
  include Figur
  def initialize(oben,unten)
    @oben = oben
    @unten = unten    
  end

  def in(p)
    @oben.in(p) || @unten.in(p)
  end

  def to_s
    "ue(o=>#{@oben} u=>#{@unten})"
  end
end

File: v1-java-vs-ruby/src/figuren/Ueberlagert.java
package figuren;

public class Ueberlagert extends Figur {

    private Figur unten;
    private Figur oben;

    public Ueberlagert(Figur oben, Figur unten) {
        super();
        this.oben = oben;
        this.unten = unten;
    }


    public String toString() {
        //return  "ue(o=>" + oben + "u=>"+ unten+ ")";
        return String.format("ue(o=>%s,u=>%s)", oben.toString(), unten.toString());
    }

    @Override
    public boolean in(Punkt p) {
        return oben.in(p) || unten.in(p);
    }
}
